### ISSUE TITLE
ubuntu 20.04.1 installed with zfs root fails to boot after installing xen-hypervisor-4.11-amd64

### ISSUE THREAD

**Issue by codesleuth7** [openzfs/zfs#10870]
Please fill in as much of the template as possible.
-->

### System information

Type | Version/Name
 --- | ---
Distribution Name	| ubuntu
Distribution Version	| 20.04.1
Linux Kernel	| 5.4.0
Architecture	| Amd
ZFS Version	| 0.8.3-1ubuntu12.1
SPL Version	| 0.8.3-1ubuntu12.1

### Describe the problem you're observing
I did some searches but failed to find anything for this problem. After installing 20.04.1 using the ZFS root option I installed Xen. When it rebooted it failed in initramfs with the error "No pool imported". I was able to boot the previous kernel. When I examined the /boot/grub/grub.cfg I found what looked like invalid "root=ZFS=" options like this:

"root=ZFS=/ROOT/ubuntu_xxxxxx"

The previous kernel that boots looks like this:

"root=ZFS=rpool/ROOT/ubuntu_xxxxxx"

After looking in /etc/default/grub and /etc/grub.d I was unable to figure out why they were different and how I could change it.

As a test saved a copy and edited /boot/grub/grub.cfg adding the rpool to each "root=ZFS=" line where rpool was missing. This time I was able to boot the Xen kernel.
### Describe how to reproduce the problem

From scratch install ubuntu 20.04.1 desktop, select ZFS as root option, then install xen-hypervisor-4.11-amd64.
It will fail to reboot because the grub root=ZFS argument is wrong.

### Include any warning/errors/backtraces from the system logs
( Unable to capture log from the initrd failure. )
The error message: "No pool imported"

**Comment by rincebrain** [comment]
You probably want to file this bug against Ubuntu's ZFS packages on Launchpad, since it's their ZFS packaging you're using.

**Comment by codesleuth7** [comment]
I could not tell who was at fault, grub, zfs, or xen. The problem seems to happen when the new xen kernel is configured into grub. I don't understand enough about how the package system adds a new kernel into grub.

**Comment by Ornias1993** [comment]
@codesleuth7 OpenZFS doesnt magically alter Grub ‘root=zfs=‘ when installing other software.

I suggest closing this, filing an issue at Ubuntu and reopen if they say OpenZFS is the cause.

Also, this looks more like a question than an actual bug.

TLDR: Please file issues on downstream first.

**Comment by codesleuth7** [comment]
https://bugs.launchpad.net/ubuntu/+source/xen/+bug/1894101
